entries/jQuery.map.xml

```diff
@@ -3,8 +3,8 @@
   <title>jQuery.map()</title>
   <signature>
     <added>1.0</added>
-    <argument name="array" type="Array">
-      <desc>The Array to translate.</desc>
+    <argument name="array" type="ArrayLikeObject">
+      <desc>The Array or an Array-like object to translate.</desc>
```

Comment:
Are we sure this is true for all versions back to 1.0?

Reply:
@timmywil I've just checked. In versions `1.0`, `1.1`, `1.2`, `1.3`, `1.4` & `1.5` `jQuery.map` is written using a for loop that would work with array-likes as well. Moreover, jQuery 1.2 implemented `jQuery.fn.map` which uses `jQuery.map` on jQuery objects which are not arrays.

jQuery `1.6` added support for iterating on objects in `jQuery.map` so it had to detect "arrays" separately. The `1.6` detection code is a bit cryptic:
```js
// jquery objects are treated as arrays
isArray = elems instanceof jQuery || length !== undefined && typeof length === "number" && ( ( length > 0 && elems[ 0 ] && elems[ length -1 ] ) || length === 0 || jQuery.isArray( elems ) )
```
but it does detect array-likes. `1.9` extracted that logic and uses:
```js
isArray = isArraylike( elems ),
```
so even the util name has "array-like" in the name.

I think we can assume that array-likes have been supported since `1.0` even if that was not the original intent.